src/stdlib/std.rs

```diff
@@ -110,6 +111,86 @@ const U8: FuncDef = func_def!(
     }
 );
 
+const LEN_BE64: FuncDef = func_def! (
+    "std::len_be64";
+    ValType::Str;
+
+    =>
+    =>
+    ValType::Str;
+
+    |mut args| {
+        let bytes: Buf = args.join_extra(b"").into();
+        let mut msg: Vec<u8> = Vec::with_capacity(bytes.len() + 5);
+
+        msg.extend((bytes.len() as u64).to_be_bytes());
+
+        msg.extend(bytes.as_ref());
+
+        Ok(Val::str(msg))
+    }
+);
+
+const LEN_BE32: FuncDef = func_def! (
+    "std::len_be32";
+    ValType::Str;
+
+    =>
+    =>
+    ValType::Str;
+
+    |mut args| {
+        let bytes: Buf = args.join_extra(b"").into();
+        let mut msg: Vec<u8> = Vec::with_capacity(bytes.len() + 5);
+
+        msg.extend((bytes.len() as u32).to_be_bytes());
+
+        msg.extend(bytes.as_ref());
+
+        Ok(Val::str(msg))
+    }
+);
+
+const LEN_BE16: FuncDef = func_def! (
+    "std::len_be16";
+    ValType::Str;
+
+    =>
+    =>
+    ValType::Str;
+
+    |mut args| {
+        let bytes: Buf = args.join_extra(b"").into();
+        let mut msg: Vec<u8> = Vec::with_capacity(bytes.len() + 5);
+
+        msg.extend((bytes.len() as u16).to_be_bytes());
+
+        msg.extend(bytes.as_ref());
+
+        Ok(Val::str(msg))
+    }
+);
+
+const LEN_U8: FuncDef = func_def! (
+    "std::len_u8";
+    ValType::Str;
+
+    =>
+    =>
+    ValType::Str;
+
+    |mut args| {
+        let bytes: Buf = args.join_extra(b"").into();
+        let mut msg: Vec<u8> = Vec::with_capacity(bytes.len() + 5);
+
+        msg.push(bytes.len() as u8);
+
+        msg.extend(bytes.as_ref());
+
+        Ok(Val::str(msg))
+    }
+);
```

Comment:
Should we be handling the overflow with a warning or anything?
or just chalk it down to user error and say nothing

Reply:
unlikely I suppose, but worth pointing out in case we accidentally use the wrong function for a field or something